### PR TITLE
Humble min laser range parameters

### LINKED ISSUE #707: Set minimum Laser Range for SLAM
#### Feature description

I am thinking of adding additional parameters to set the minimum laser range (as it now support the `max_laser_range`). I have a use cases where my robot goes under a trolley and lurk / pull the trolley around. I need to do lifelong mapping due to the highly dynamic environment. I got a problem where my laserscan data read the trolley legs and add it into the map causing the robot could not move anywhere.

#### Implementation considerations

It could be solved either by:
1. Add `min_laser_range` parameters for slam_toolbox to ignore laserscan readings that are lower than the value
2. Filter my laserscan data out manualy 

I personally prefer the first option since I might need the second one for my navigation just in case I have rectangle instead of square.

### PR DESCRIPTION
**Pull request by mich1342**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | [#707 ](https://github.com/SteveMacenski/slam_toolbox/issues/707)|
| Primary OS tested on | Ubuntu 22.04|
| Robotic platform tested on | Linorobot2 Gazebo Simmulation, My customized ROS2 robot |

---

## Description of contribution in a few bullet points

* I added `min_laser_range` parameter to limit the minimum range being used by the SLAM to create the map similar to `max_laser_range` parameter
* I add short description in the README.MD about the new parameter
* I add the parameter to all of the sample config file with value of 0 that will trigger the use of laser scan message minimum range

## Description of documentation updates required from your changes

* Added new parameter, so need to add that to default configs and documentation page (I have done it in the README part)

---

### CODE REVIEW

File: src/laser_utils.cpp [hunk 1]
```diff
@@ -26,176 +26,212 @@
 namespace laser_utils
 {
 
-LaserMetadata::LaserMetadata()
```
Comment (SteveMacenski) [inline src/laser_utils.cpp]: Revert file reformatting so I can see what you changed

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: src/laser_utils.cpp (hunk 1 above)
Comment (mich1342) [reply src/laser_utils.cpp]: Hi @SteveMacenski ,

Sorry for the reformatting issue. I have revert all reformatting leaving the main changes only. Thankyou

File: src/laser_utils.cpp [hunk 2]
```diff
@@ -99,7 +99,28 @@ karto::LaserRangeFinder * LaserAssistant::makeLaser(const double & mountingYaw)
     karto::LaserRangeFinder_Custom, karto::Name("Custom Described Lidar"));
   laser->SetOffsetPose(karto::Pose2(laser_pose_.transform.translation.x,
     laser_pose_.transform.translation.y, mountingYaw));
-  laser->SetMinimumRange(scan_.range_min);
+
+  double min_laser_range = 25;
```
Comment (SteveMacenski) [inline src/laser_utils.cpp]: `0.0`

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: src/laser_utils.cpp (hunk 2 above)
Comment (mich1342) [reply src/laser_utils.cpp]: Thanks, have updated it in the latest commit

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Please open a PR with this for `ros2` and I can merge the pair!

**Comment by mich1342** [comment]
> Please open a PR with this for `ros2` and I can merge the pair!

Done in #714

**Pull request merged**